cpu/stm32_common/periph/gpio.c

```diff
@@ -253,6 +257,30 @@ void isr_exti(void)
 }
 #endif /* MODULE_PERIPH_GPIO_IRQ */
 
-#else
+#ifdef MODULE_PERIPH_GPIO_ABC
+static inline __attribute__((always_inline)) void _delay(int32_t loops)
+{
+    __asm__ volatile(
+        "0:" "SUBS %[iterator], #1;"
+        "BPL 0b;"
+        : [iterator]"+r"(loops)
+    );
+}
+
+void gpio_set_for(gpio_t pin, int delay)
+{
+    _port(pin)->BSRR = (1 << _pin_num(pin));
```

Comment:
STM32 support bit-banding, so you can be even faster

```C
#define BITBAND_SRAM_REF   0x20000000
#define BITBAND_SRAM_BASE  0x22000000

#define BITBAND_SRAM(a,b)  (*((volatile uint8_t *) ((BITBAND_SRAM_BASE + ((uintptr_t) (a)-BITBAND_SRAM_REF)*32 + (b*4)))))  /* Translate SRAM address to Bit-Band space */
#define BITBAND_PERI_REF   0x40000000
#define BITBAND_PERI_BASE  0x42000000
#define BITBAND_PERI(a,b)  (*((volatile uint8_t *) ((BITBAND_PERI_BASE + ((uintptr_t) (a)-BITBAND_PERI_REF)*32 + (b*4))))) /* Translate PERIPHERAL address  to Bit-Band space */

#define BIT_SET_HW(val, bit)    (BITBAND_PERI(&(val), (bit)) = 1)
#define BIT_DEL_HW(val, bit)    (BITBAND_PERI(&(val), (bit)) = 0)
#define BIT_CHK_HW(val, bit)    (BITBAND_PERI(&(val), (bit)))
```

```suggestion
    BIT_SET_HW(_port(pin)->BSRR, _pin_num(pin));
```

Reply:
Hm now I wonder if this is still faster with all the added math, it would sure be if all parameters were compile-time constant :thinking: